### PR TITLE
Main screen props

### PR DESCRIPTION
**Pull request by AlexeyDemedetskiy**
Some earlly thoughts.

No props for rendering links yet.

### CODE REVIEW

File: Main Screen/OverviewViewControllerProps.swift [hunk 1]
```diff
@@ -0,0 +1,49 @@
+extension OverviewViewController {
+    struct Props {
+        let actions: [Action]
+        let newAction: NewEntry
+        let states: [State]
```
Comment (MaximBazarov) [inline Main Screen/OverviewViewControllerProps.swift]: new line here would be cool

File: Main Screen/OverviewViewControllerProps.swift [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+        let newState: NewEntry
+        
+        let props: [Props]
+        let newProps: NewEntry
+
```
Comment (MaximBazarov) [inline Main Screen/OverviewViewControllerProps.swift]: Could you please elaborate on this idea?

As I get it:
- we send `AddProps` action
- `newProps` state gets updated
- when we have `case invalid(reason: String)` we show an error
- when `case valid(save: Command)` we make a button "save" active? what would the command be like?

are we going to compare an entity name with the current props name in the VC? or why do we need it in `NewEntry`?

File: Main Screen/OverviewViewControllerProps.swift [hunk 3]
```diff
@@ -0,0 +1,49 @@
...
+            let fields: [Field]
+            let connectInput: Command?
+        }
+        
+        struct Field {
+            let name: String
+            let type: String
+        }
+        
+        struct NewEntry {
```
Comment (MaximBazarov) [inline Main Screen/OverviewViewControllerProps.swift]: could you please elaborate on commands?

File: Main Screen/OverviewViewControllerProps.swift (hunk 2 above)
Comment (AlexeyDemedetskiy) [reply Main Screen/OverviewViewControllerProps.swift]: `AddProps` is the last command in a sequence of adding new props.

At first, it needs to be activated by tapping a + button. The name should be updated. If the name is valid (unique and not empty) save button activates.

State of this process is independent and standalone state with its own set of actions.

**Pull request merged**